src/riak_kv_delete.erl

```diff
@@ -189,11 +199,61 @@ get_w_options(Bucket, Options) ->
 %%       these values but a 2-tuple list, namely, 'sloppy_quorum'
 %%       cannot appear as a proplist-style property but must always
 %%       appear as {sloppy_quorum, boolean()}.
-
 extract_passthru_options(Options) ->
     [Opt || {K, _} = Opt <- Options,
             K == sloppy_quorum orelse K == n_val].
 
+
+%% riak_kv_sweeper callbacks
+participate_in_sweep(Index, _Pid) ->
+    InitialAcc = 0,
+    {ok, reap_fun(Index), InitialAcc}.
+
+reap_fun(Index) ->
+    fun({_BKey, RObj}, Acc, _Opt) ->
+            case riak_kv_util:obj_not_deleted(RObj) of
+                undefined ->
+                    maybe_reap(RObj, Index, Acc);
+                _ ->
+                    {ok, Acc}
+            end
+    end.
+
+maybe_reap(RObj, Index, Acc) ->
+    case obj_outside_grace_period(RObj) of
+        true ->
+            riak_kv_stat:update({reap_tombstone, Index}),
+            {deleted, Acc};
+        false ->
+            {ok, Acc}
+    end.
+
+obj_outside_grace_period(RObj) ->
+    obj_outside_grace_period(RObj, app_helper:get_env(riak_kv, tombstone_grace_period)).
+obj_outside_grace_period(_RObj, false) ->
+	false;
+obj_outside_grace_period(RObj, TombstoneGracePeriod) ->
+	Now = os:timestamp(),
+
+	Contents = riak_object:get_contents(RObj),
+	case [{M, V} || {M, V} <- Contents,
```

Comment:
Good catch. Fixed! (Although I used `lists:all` instead of `lists:any` since I believe we only want to reap the object if all the tombstones are outside the grace period, not just one.)